packages/firebase_analytics/firebase_analytics/test/observer_test.dart

```diff
@@ -147,6 +145,31 @@ void main() {
   });
 }
 
-class MockFirebaseAnalytics extends Mock implements FirebaseAnalytics {}
+class MockFirebaseAnalytics extends Mock implements FirebaseAnalytics {
+  @override
+  Future<void> setCurrentScreen(
+      {required String? screenName, String? screenClassOverride = 'Flutter'}) {
+    return super.noSuchMethod(
+      Invocation.method(#setCurrentScreen, [],
+          {#screenName: screenName, #screenClassOverride: screenClassOverride}),
+      returnValue: neverEndingFuture<void>(),
```

Comment:
Why `neverEndingFuture`? Using `Future<void>.value()` passes all tests. If it is really needed, `Completer().future` can be used.

Reply:
But don't all tests already succeed? 🤔 

I've oriented myself here on the @rrousselGit implementation for `firebase_auth` ☺️
https://github.com/FirebaseExtended/flutterfire/blame/master/packages/firebase_auth/firebase_auth/test/mock.dart

Reply:
Yes, this passes tests but using `Future<void>.value()` or `Completer<void>().future` is much simple.

Reply:
Ah for `void`, indeed `Future<void>.value` is a better solution. The `neverEdingFuture<void>` are there by mistake you could say.